include/pybind11/detail/common.h

```diff
@@ -695,6 +695,19 @@ using expand_side_effects = bool[];
 #define PYBIND11_EXPAND_SIDE_EFFECTS(PATTERN) (void)pybind11::detail::expand_side_effects{ ((PATTERN), void(), false)..., false }
 #endif
 
+// Thread state manipulation C API should not be called while Python runtime is finalizing.
+// For more detail see https://docs.python.org/3.7/c-api/init.html#c.PyEval_RestoreThread
```

Comment:
I'm not entirely convinced this should be here (instead of at the point where it's used), if you're re-purposing this `is_finalizing` to be more generic?

Reply:
I don't have a strong preference on the method location, moved it back to `pybind11.h` (haven't realized that you guys are OK with mixing several namespaces in the same header file)

Reply:
Sorry if I didn't make myself understood correctly. I was talking about the two lines of documentation (as mentioned before: https://github.com/pybind/pybind11/pull/2657#discussion_r525238495). If you preferred, the function could stay in `common.h`.